### PR TITLE
Adding history listener to update persisted queries for nav

### LINKED ISSUE #5684: Navigation: Make sure persisted queries are present on wc-admin links
Persisted queries are query parameters that remain when navigating from report to report. They include things like `period` so users can view reports from the same vantage point.

https://github.com/woocommerce/woocommerce-admin/blob/2a382d72b85ada88df28005f91910294909587c6/packages/navigation/src/index.js#L42-L51

In the old nav, we update the links so that those query parameters are present when they should be.

https://github.com/woocommerce/woocommerce-admin/blob/2a382d72b85ada88df28005f91910294909587c6/client/layout/controller.js#L214

The new Nav needs to also have those parameters present.

### Deliverables

* When looking at a report with any of the persisted queries in the url, any link to another report or overview page should also include those parameters.

### PR DESCRIPTION
**Pull request by joelclimbsthings**
Fixes #5684

The persisted queries haven't been persisting when using the new navigation. The logic to include this is already present, but there is nothing triggering a re-render in the plugin component when the history changes so the logic isn't utilized. 

This PR adds the history listener and triggers a state change when the persisted queries are updated.

### Detailed test instructions:

-  Check out branch
- Ensure that the new navigation is enabled.
- Go to an _Analytics_ report page and apply a different date range.
- Then navigate to a different report page (without a refresh), and ensure that the new date range persists.
- Disable the new navigation and do the same, ensuring that the old nav works as always.

### CODE REVIEW

**Review by becdetat — APPROVED** [review]
Very neat and works well 🚢

**Review by joshuatf — COMMENTED** [review]
Nice job, Joel!  I'm not sure why this didn't work before when I tried it, but it's working well in this PR 😄 

I'm tempted to go with this route because we don't risk breaking things in WCA.  We're directly manipulating the DOM still in the old WP menu, but without doing so we lose the ability to get persisted queries on opening a new tab (unless we begin caching those somewhere).  I feel like that's a separate issue to tackle from the new nav one and we could look at caching or using virtual dom for the old WP menu in a follow-up.

Will wait for more buy-in from the @woocommerce/isotope before approving, but this LGTM as is for me.

**Review by psealock — COMMENTED** [review]
I love the simplicity, well done @joelclimbsthings 

This is testing well for me and the code looks good. 

As you mentioned in your summary in the original issue, these query parameters are also being applied to excluded screens. I think you can use the same logic to conditionally apply the persisted queries so they aren't added to items like Customers.

https://github.com/woocommerce/woocommerce-admin/blob/b129f3c25a0b40577fc1e31a4871e041d5db32d5/client/layout/controller.js#L251-L256

I also noticed our `Link` component doesn't respect OSX cmd + click shortcut to open links on a new window. I created an issue  https://github.com/woocommerce/woocommerce-admin/issues/5916

**Comment by joelclimbsthings** [comment]
Thanks all! I've centralized in the excluded screens logic as mentioned by @psealock  and added it to the implementation for the new navigation.

File: client/layout/navigation.js [hunk 1]
```diff
@@ -6,18 +6,32 @@ import {
...
 const NavigationPlugin = () => {
+	const [ persistedQuery, setPersistedQuery ] = useState(
+		getPersistedQuery()
+	);
+
+	// Update the persisted queries when history is updated
+	useEffect( () => {
+		return addHistoryListener( () => {
+			setTimeout( () => setPersistedQuery( getPersistedQuery() ), 0 );
+		} );
```
Comment (psealock) [inline client/layout/navigation.js:32]: In testing I see that going to `Customers` and then back to a report looses the persisted query. Do you think we could skip this function call to `setPersistedQuery` when on the Customers page or another excluded page?

Comment (joelclimbsthings) [reply client/layout/navigation.js:41]: Great call @psealock . Just committed this change, and ended up refactoring a few things as well.

File: packages/navigation/src/index.js [hunk 2]
```diff
@@ -46,6 +48,19 @@ export const getPersistedQuery = ( query = navUtils.getQuery() ) => {
...
+	applyFilters( TIME_EXCLUDED_SCREENS_FILTER, [
+		'stock',
+		'settings',
+		'customers',
+		'homescreen',
+	] );
+
 /**
  * Get an array of IDs from a comma-separated query parameter.
  *
```
[the remark this answers is not in the document]
Comment (joelclimbsthings) [reply packages/navigation/src/index.js:78]: Done. Do I need to bump the version as well, or is that done through the release?

**Comment by joelclimbsthings** [comment]
Thanks for the review @psealock . I've addressed your feedback, and ended up refactoring a couple things in the process :+1:

File: packages/navigation/src/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (psealock) [reply packages/navigation/src/index.js:78]: We can do it as part of the release

**Review by psealock — APPROVED** [review]
Looking great @joelclimbsthings ! Thank you. This is testing well for me and the code looks good. I like the changes and additional refactor too. Nice job delivering a robust solution to a complicated problem!

**Pull request merged**